src/shared/hooks/useNS.ts

```diff
@@ -1,20 +1,29 @@
 import { useRouteMatch } from 'react-router-dom';
-import { allSearchResultLocations } from '../../app/config/urls';
+import { allIDMappingTargetLocations } from '../../app/config/urls';
 
 import { Namespace } from '../types/namespaces';
+import { IDMappingNamespace } from '../../tools/id-mapping/types/idMappingServerParameters';
 
-const useNS = (): Namespace | undefined => {
-  const match = useRouteMatch<{ namespace: Namespace }>(
-    allSearchResultLocations
-  );
+const useNS = (): [Namespace, boolean] => {
```

Comment:
Could we keep `IDMappingNamespace` out of `Namespace` by having `useNS` look at the url to determine if it's at a tool location (id mapping or peptide search) and returning `[namespace, tool]` in that instance.

Reply:
There is a particularity to `id-mapping` in the sense that it is _also_ a namespace (when the target is just an ID). This means it has its own definition for default columns, primary key, column config etc...
However if this is useful for peptide search, I could return `secondaryNS` instead of a boolean:
- `/id-mapping/12345/uniprot` -> `[uniprot, idmapping]`
- `/id-mapping/12345/uniref` -> `[uniref, idmapping]`
- `/id-mapping/12345` -> `[idmapping, undefined]`
- `/peptidesearch/12345` -> `[uniprot, peptidesearch]`

Reply:
`/id-mapping/uniprot/12345`*
`/id-mapping/uniref/12345`*

Reply:
Thanks @aurel-l , edited

Reply:
I've just merged in this branch into my peptide search branch and everything I've implemented so far works fine so need to change your code.